bundles/framework/printout/PrintoutHandler.js

```diff
@@ -0,0 +1,317 @@
+        const payload = { customStyles };
+        let fileName = params.pageTitle || 'print';
+        const format = FORMAT_OPTIONS.find(format => format.mime === params.format);
+        if (format) {
+            fileName += '.' + format.name;
+        }
+
+        this.updateState({
+            loading: true
+        });
+        const successCb = blob => {
+            if (navigator.msSaveOrOpenBlob) {
+                navigator.msSaveOrOpenBlob(blob, fileName);
+            } else {
+                const url = window.URL.createObjectURL(blob);
+                const elem = document.createElement('a');
+                elem.href = url;
+                elem.download = fileName;
+                document.body.appendChild(elem);
+                elem.click();
+                document.body.removeChild(elem);
+                window.URL.revokeObjectURL(url);
+            }
+            this.updateState({
+                loading: false
+            });
+        };
+        const errorCb = error => {
+            Oskari.log('BasicPrintout').error(error);
+            this.updateState({
+                loading: false
+            });
+            Messaging.error(Oskari.getMsg('Printout', 'BasicView.error.saveFailed'));
+        };
+        this.instance.getService().fetchPrint(printUrl, payload, successCb, errorCb);
+    }
+
+    getSelectedCustomStyles () {
+        let customStyles = {};
+        const selectedLayers = this.sandbox.findAllSelectedMapLayers();
+
+        selectedLayers.forEach(l => {
+            const style = l.getCurrentStyle();
+            if (typeof style.isRuntimeStyle === 'function' && style.isRuntimeStyle()) {
+                customStyles[l.getId()] = style.getFeatureStyle();
+            }
+        });
+        return customStyles;
+    }
+
+    isLandscape (pageSize) {
+        const ps = pageSize || this.state.size;
+        const opt = SIZE_OPTIONS.find(o => o.value === ps);
+        return opt ? opt.landscape : SIZE_OPTIONS[0].landscape;
+    }
+
+    isTimeSeriesActive () {
+        const hasLayers = this.instance.sandbox.findAllSelectedMapLayers().filter(l => l.getAttributes().times).length > 0;
+        return hasLayers && !!this.timeseriesPlugin;
+    }
+
+    getUrlForPreview (scaledWidth) {
+        const pageSize = this.state.size;
+        const map = Oskari.getSandbox().getMap();
+        const baseLayer = this.mapModule.getBaseLayer();
+
+        let mapLayers = '';
+        if (baseLayer) {
+            mapLayers = baseLayer.getId() + ' ' +
+            baseLayer.getOpacity() + ' ' +
+            baseLayer.getCurrentStyle().getName();
+        }
+
+        let url = Oskari.urls.getRoute('GetPrint') +
+            '&format=image/png' +
+            '&pageSize=' + pageSize +
+            '&resolution=' + map.getResolution() +
+            '&srs=' + map.getSrsName() +
+            '&coord=' + map.getX() + '_' + map.getY() +
+            '&mapLayers=' + mapLayers;
```

Comment:
You can pass params to getRoute as an object:
```suggestion
        let url = Oskari.urls.getRoute('GetPrint', {
         format: 'image/png',
         ...etc})
```